addons/payment_authorize/models/payment_acquirer.py

```diff
@@ -39,6 +39,13 @@ class PaymentAcquirer(models.Model):
         required_if_provider='authorize',
     )
 
+    def write(self, vals):
+        for acquirer in self:
+            if acquirer.provider == 'authorize' and not acquirer.authorize_currency_id and vals.get('state') in ['enabled', 'test']:
+                if not 'authorize_currency_id' in vals:
```

Comment:
This only works when, in the same `write` call, the acquirer is enabled, and the currency was not set beforehand. If you unset the currency while the acquirer is enabled, for example, no error will be raised.

Instead of dealing with `write` values, call `super` first then use a `filtered` on the up-to-date `self` and raise the error if the currency is not set while the acquirer is enabled. That will fix the problem and be simpler at the same time.